src/libmongoc/tests/TestSuite.c

```diff
@@ -137,6 +137,7 @@ TestSuite_Init (TestSuite *suite, const char *name, int argc, char **argv)
    suite->flags = 0;
    suite->prgname = bson_strdup (argv[0]);
    suite->silent = false;
+   _mongoc_array_init (&suite->match_patterns, sizeof (char *));
```

Comment:
Note, now `test-libmongoc` can be invoked with multiple occurrences of `-l <pattern>`. This was necessary to invoke with the multiple subset of tests we need to run against a load balanced deployment.

I also added an alias of `--match`.